### PR TITLE
Feature/recovery password

### PR DESCRIPTION
**Pull request by Timak31**
Rewrote the logic of sending the email. Changed sending an email with confirmation. Now all html e-mail pages are in resources/templates like .html files (previously it was an HTML string in the variable). If the user forgot the password, he clicks the link "forgot my password" after then his redirect on the recover's page where he enters the email. The server gets the email then sending html page with a recover's link after clicking the link the user gets a letter with his new (generated) password.

### CODE REVIEW

**Comment by Timak31** [comment]
Rewrote the logic of sending the email. Changed sending an email with confirmation. Now all html e-mail pages are in resources/templates like .html files (previously it was an HTML string in the variable). If the user forgot the password, he clicks the link "forgot my password" after then his redirect on the recover's page where he enters the email. The server gets the email then sending html page with a recover's link after clicking the link the user gets a letter with his new (generated) password.

File: client/src/pages/forgot-password/ForgotPassword.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public render() {
+        return (
+            <div className='container '>
+                <div className="jumbotron">
+                    <div className='container '>
+                        <h1>Recovery password page</h1>
+                        <hr />
+                        <div className="card-body">
+                            <form id='recovery-dialog' onSubmit={this.handleClick} >
+                                <input className="form-control" type="text" onChange={this.handle.bind(this)}/>
```
Comment (Chubasic) [inline client/src/pages/forgot-password/ForgotPassword.tsx]: type="email"

File: client/src/pages/forgot-password/SendRecoveryEmail.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        super(props);
+        this.token = '';
+        console.debug(this.props)
+    }
+
+    public componentWillMount() {
+        this.token = window.location.toString();
+    }
+
+    public async componentDidMount() {
```
Comment (Chubasic) [inline client/src/pages/forgot-password/SendRecoveryEmail.tsx:18]: Should the user authorize after this request? (No)

Comment (Chubasic) [reply client/src/pages/forgot-password/SendRecoveryEmail.tsx:18]: call your request here

File: client/src/pages/forgot-password/SendRecoveryEmail.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import {Redirect} from "react-router";
+
+interface SendNewPasswordPropsTypes {
+    context: {
+        authorized: Auth['authorized'];
+        authorize: Auth['authorize'];
+    };
+}
+
+export class SendRecoveryEmail extends Component<SendNewPasswordPropsTypes, any> {
```
Comment (Chubasic) [inline client/src/pages/forgot-password/SendRecoveryEmail.tsx]: That is very bad name for interface/type

File: server/goadventures/src/main/java/io/softserve/goadventures/auth/service/JWTService.java [hunk 4]
```diff
@@ -1,49 +1,64 @@
...
+     * Create refresh token from email
+     */
+    public String refreshToken(String email) {
+        return JWT.create()
+                .withSubject(email)
+                .withExpiresAt(new Date(System.currentTimeMillis() + REFRESH_TIME))
+                .sign(HMAC256(REFRESH_SECRET.getBytes()));
+    }
 
+    public String parseRedreshToken(String token) {
```
Comment (Chubasic) [inline server/goadventures/src/main/java/io/softserve/goadventures/auth/service/JWTService.java]: Typo "parseRedreshToken" :)

File: server/goadventures/src/main/java/io/softserve/goadventures/auth/controller/AuthController.java [hunk 5]
```diff
@@ -15,258 +16,190 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (Chubasic) [inline server/goadventures/src/main/java/io/softserve/goadventures/auth/controller/AuthController.java:190]: :( :-1:

File: client/src/pages/forgot-password/SendRecoveryEmail.tsx (hunk 3 above)
Comment (Timak31) [reply client/src/pages/forgot-password/SendRecoveryEmail.tsx]: Seriously??? It's your template.
`interface ConfirmPropsTypes {
  context: {
    authorized: Auth['authorized'];
    authorize: Auth['authorize'];
  };
}

export class Confirm extends Component<ConfirmPropsTypes, any> {
  private url: string;
`

**Pull request merged**